src/components/Autocomplete/README.md

```diff
@@ -301,6 +301,123 @@ class AutocompleteExample extends React.Component {
 }
 ```
 
+### Autocomplete with lazy loading
+
+```jsx
+class AutoCompleteLazyLoadExample extends React.Component {
+  paginationInterval = 25;
+
+  options = Array.from(Array(1000)).map((_, index) => ({
+    value: `rustic ${index}`,
+    label: `Rustic ${index}`,
+  }));
+
+  state = {
+    selected: [],
+    inputText: '',
+    options: this.options,
+    visibleOptionIndex: this.paginationInterval,
+  };
+
+  render() {
+    const {visibleOptionIndex, selected, options} = this.state;
+    const textField = (
+      <Autocomplete.TextField
+        onChange={this.updateText}
+        label=""
+        value={this.state.inputText}
+        placeholder="Vintage, cotton, summer"
+      />
+    );
+
+    const optionList = options.slice(0, visibleOptionIndex);
+    const selectedTagMarkup =
+      selected.length > 0 ? (
+        <Stack spacing="extraTight">{this.renderTags()}</Stack>
+      ) : null;
+
+    return (
+      <Stack vertical>
+        {selectedTagMarkup}
+        <Autocomplete
+          allowMultiple
+          options={optionList}
+          selected={this.state.selected}
+          textField={textField}
+          onSelect={this.updateSelection}
+          listTitle="Suggested Tags"
+          onLoadMoreResults={this.handleLoadMoreResults}
+        />
+      </Stack>
+    );
+  }
+
+  handleLoadMoreResults = () => {
+    const {visibleOptionIndex} = this.state;
+    const nextVisibleOptionIndex = visibleOptionIndex + this.paginationInterval;
+    if (nextVisibleOptionIndex <= this.options.length - 1) {
+      this.setState({visibleOptionIndex: nextVisibleOptionIndex});
+    }
+  };
+
+  updateText = (newValue) => {
+    this.setState({inputText: newValue});
+    this.filterAndUpdateOptions(newValue);
+  };
+
+  removeTag = (tag) => () => {
+    const newSelected = this.state.selected;
+    newSelected.splice(newSelected.indexOf(tag), 1);
+    this.setState({selected: newSelected});
+  };
+
+  renderTags = () => {
+    return this.state.selected.map((option) => {
+      let tagLabel = '';
+      tagLabel = option.replace('_', ' ');
+      tagLabel = titleCase(tagLabel);
+      return (
+        <Tag key={`option${option}`} onRemove={this.removeTag(option)}>
+          {tagLabel}
+        </Tag>
+      );
+    });
+  };
+
+  filterAndUpdateOptions = (inputString) => {
+    if (inputString === '') {
+      this.setState({options: this.options});
+
+      return;
+    }
+
+    const filterRegex = new RegExp(inputString, 'i');
+    const resultOptions = this.options.filter((option) =>
+      option.label.match(filterRegex),
+    );
+    let endIndex = resultOptions.length - 1;
+    if (resultOptions.length === 0) {
+      endIndex = 0;
+    }
+    this.setState({
+      options: resultOptions,
+    });
```

Comment:
Having `and` in the function name feels like a good sign this should be two methods. I'd suggest pulling the `setState` out into its own function `updateOptions(newOptions)` and call it here.